src/Geralt/Crypto/IncrementalXChaCha20Poly1305.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Security.Cryptography;
+using static Interop.Libsodium;
+
+namespace Geralt;
+
+public sealed class IncrementalXChaCha20Poly1305 : IDisposable
+{
+    private crypto_secretstream_xchacha20poly1305_state _state;
+    private bool _forDecryption;
+
+    /// <summary>
+    /// Size of the authentication tag which gets appended to each ciphertext chunk.
+    /// </summary>
+    public const int TagSize = crypto_secretstream_xchacha20poly1305_ABYTES;
+
+    /// <summary>
+    /// Size of the header, which gets created when initializing for encryption and must be provided back when initializing for decryption.
+    /// </summary>
+    public const int HeaderBytes = crypto_secretstream_xchacha20poly1305_HEADERBYTES;
+
+    /// <summary>
+    /// Size of the secret key used to encrypt and decrypt a message.
+    /// </summary>
+    public const int KeySize = crypto_secretstream_xchacha20poly1305_KEYBYTES;
+
+    /// <summary>
+    /// No information is added to the nature of the message.
+    /// </summary>
+    public const byte TagMessage = crypto_secretstream_xchacha20poly1305_TAG_MESSAGE;
+
+    /// <summary>
+    /// Mark the end of the stream.
+    /// </summary>
+    public const byte TagFinal = crypto_secretstream_xchacha20poly1305_TAG_FINAL;
+
+    /// <summary>
+    /// Mark the end of a sequence of messages, but not the end of the stream.
+    /// </summary>
+    public const byte TagPush = crypto_secretstream_xchacha20poly1305_TAG_PUSH;
+
+    /// <summary>
+    /// "Forget" the current secret key and derive a new one.
+    /// </summary>
+    public const byte TagReKey = crypto_secretstream_xchacha20poly1305_TAG_REKEY;

```

Comment:
I'd rename this to `TagRekey`. Furthermore, these const bytes might be better of as an enum to prevent the user passing other bytes in.